general/react_dnd.md

```diff
@@ -0,0 +1,170 @@
+# React Dnd
+
+react-dnd, wether is a good and very flexible drag and drop library for react, has one of the worst documentations ever. Anyway this is not intended to be a full replacement of reat-dnd docs but just some guides and tips to use it on a react application via hooks. For full API reference visit https://react-dnd.github.io/react-dnd/
+
+## Backend Provider
+
+react-dnd requires your dnd-ready apps to be wrapped inside what they call a backend provider in order to work. There're two main backends the library provides: HTML5, for regular mouse events, and Touch, for touch events.
+
+```
+import Backend from 'react-dnd-html5-backend'
+import { DndProvider } from 'react-dnd'
+
+export default function MyReactApp() {
+  return (
+    <DndProvider backend={Backend}>
+      /* your drag-and-drop application */
+    </DndProvider>
+  )
+}
+```
+
+## useDrag
+
+First thing you need to configure is your draggable elements using useDrag hook.
+
+```
+const [{isDragging}, drag, preview] = useDrag({
+		collect: monitor => ({
+			isDragging: !!monitor.isDragging(),
+		}),
+		item: {
+            id, 
+            type: ACCEPTING_TYPES.ITEM
+        },
+	});
+```
+
+### Return value array
+
+`useDrag` returns an array of three values:
+
+- `Index 0`: An object containing collected properties from the collect function. If no collect function is defined, an empty object is returned.
+- `Index 1`: A connector function for the drag source. This **must** be attached to the draggable portion of the DOM.
+- `Index 2`: A connector function for the drag preview. This may be attached to the preview portion of the DOM in case you want to overwrite the default drag preview behavior.
+
+### Hook config object (most important properties)
+
+- `item`: Required. This is the only information about the draggable item that will be available for the drop or custom drag layer objects, so you need to place here any information you'll need on your drop logic.
+
+    `item.type` is also required, only drop targets registered for the same type will react to this drag.
+
+- `collect`: Optional. This is a function that receives `monitor` and `props`, and must return a plain object that you'll receive as the fist array element of the hook return value.
+
+### Connect drag with DOM element
+
+You can connect drag with an element in two differente ways:
+
+- Setting `drag` returned parameter as your element's `ref`.
+
+```
+return (
+    <div ref={drag}>
+        ...
+    </div>
+);
+```
+
+- Creating your own `ref`, calling `drag` returned parameter passing it out this ref and setting your element's ref with it.
```

Comment:
This sentence doesn't sound quite right. Suggest either:

> Creating your own ref, passing it to `drag()`, and using the ref on the element.

(Note that you don't need backticks around "ref"; in [the React docs](https://reactjs.org/docs/refs-and-the-dom.html), "Ref" is just a normal noun — the variable name itself doesn't matter).

Or just:

> Creating your own ref.

(Because the code same makes it very obvious what you should do with it.)